Examples/ByFeature/FunctionalInjection/Logger.fs

```diff
@@ -0,0 +1,23 @@
+﻿module Logger
+
+open TickSpec
+open System
+
+type LogMessages = string list
+
+type LoggerContext = { Messages: LogMessages }
+
+let [<BeforeScenario>] setup () = 
+    { Messages = "Before scenario" |> List.singleton }
+
+let [<BeforeStep>] beforeStep (previousMessages: LoggerContext) =
+    { Messages = "Before step" :: previousMessages.Messages  }
+
+let [<AfterStep>] afterStep (previousMessages: LoggerContext) =
+    { Messages = "After step" :: previousMessages.Messages  }
+
+let [<AfterScenario>] afterScenario (previousMessages: LoggerContext) =
+    let allMessages =
+        "After scenario" :: previousMessages.Messages
+        |> List.rev
```

Comment:
There's no value in having the interstitial reversed state manifest as a linked list of nodes, so:
allMessages |> Seq.rev |> Seq.iter Console.Writeline
(I'd prob write `for m in Seq.rev allMessages do Console.WriteLine m` myself)

Reply:
Good point.